skyfield/sgp4lib.py

```diff
@@ -87,6 +103,41 @@ def gcrs(self, jd):
         g.sgp4_error = error
         return g
 
+    @property
+    def mean_motion(self):
+        # revolutions per second
+        return self._sgp4_satellite.no / 60
+
+    @property
+    def orbital_period(self):
+        return timedelta(seconds=2 * pi / self.mean_motion)
+
+    @takes_julian_date
+    def orbital_speed(self, jd):
+        """Return orbital speed of this satellite in meters per second.
+        """
+        elevation = self.gcrs(jd).distance().m
+        semi_major_axis = self.semi_major_axis.m
+        return sqrt(GE * ((2 / elevation) - (1 / semi_major_axis)))
+
+    @takes_julian_date
+    def over_location(self, jd):
+        """Return a Topos instance for the point on the Earth over which
+        this satellite will be at the given date.
+
+        >>> from skyfield.api import earth, JulianDate
+        >>> tle = ("ISS (ZARYA)\n"
+        ... "1 25544U 98067A   15058.48161588  .00023857  00000-0  35618-3 0  9991\n"
+        ... "2 25544  51.6478 271.9610 0008043  54.4849  18.6646 15.54887163930964")
+        >>> iss = earth.satellite(tle)
+        >>> topos = iss.over_location(JulianDate(utc=(2015, 2, 27, 22, 22, 0)))
+        >>> topos.latitude
+        <Angle 49deg 19' 48.3">
+        >>> topos.longitude
+        <Angle -155deg 37' 41.8">
```

Comment:
I would like to see this test go one step farther: now that you have a `Topos`, ask it to observe the satellite, and demonstrate that the `Topos` thinks that (a) the satellite has an altitude of exactly (or very nearly) 90° above the horizon, and (b) states the range to the satellite is the same as the elevation that is returned by your elevation function. You can get the altitude angle and range by running `.altaz()`, I think. By doing a full “round trip” from the satellite to the `Topos` and then back the other way, we will learn whether all the math is consistent. Thanks!